### ISSUE TITLE
Buildstream fails to spot bad project.conf

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#1019]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/1019)
In GitLab by [[Gitlab user @pointswaves]](https://gitlab.com/pointswaves) on May 8, 2019, 16:12

## Summary

Some mal formed project.conf files cause bst to think there is no project and ask if you want to make a new one.
but if you decline they then tell you about your project.conf file being bad.

## Steps to reproduce

I don't have a minimal reproducible test but something like this https://gitlab.com/pointswaves/bootable-images/blob/bug/project.conf#L23 should be able to be simplified to a test case.

## What is the current bug behavior?

bst should report the issue with the project.conf and not think there is no porject

## What is the expected correct behavior?
report a problem with the project.conf

----

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on May 8, 2019, 16:22

Hi, we do pretty strong validation and error reporting for any loaded YAML fields, so we prefer to treat any problems in this area as individual, fixable issues (as opposed to a single general issue).

Can you please specify exactly what field caused an improper error, or what was the nature of the mis-reported error ?

Thanks for taking the time to report this !

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @abderrahimk]](https://gitlab.com/abderrahimk) on May 8, 2019, 17:22

I noticed this before: when there is a syntax error in the project.conf, bst acts as if the file doesn't exist and asks if you want to create a project.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @pointswaves]](https://gitlab.com/pointswaves) on May 8, 2019, 18:02

changed the description

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @pointswaves]](https://gitlab.com/pointswaves) on May 8, 2019, 18:03

Sorry, I meant to link the line its missing a `"` https://gitlab.com/pointswaves/bootable-images/blob/bug/project.conf#L23, I have updated the description to point to the bad line.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @pointswaves]](https://gitlab.com/pointswaves) on May 8, 2019, 18:07

Here is a copy and paste:
```
projects  pithings  streamimages  bug  4  $  bst build DiskImage.bst 
A project was not detected in the directory: /home/will/projects/pithings/streamimages

Would you like to create a new project here ? [y/N]: n

Error loading project: Malformed YAML:

mapping values are not allowed here

  in "/home/will/projects/pithings/streamimages/project.conf", line 26, column 26
```

I am not complaining about the error message although it is a bit cryptic but about the order. Shouldn't it complain about project.conf before offering to create a new project?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @pointswaves]](https://gitlab.com/pointswaves) on May 8, 2019, 18:15

I also should have mentioned that this is using bst1.2 as i am junction-ing from freedesktop-sdk 18.08

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on May 8, 2019, 18:15

It should not offer to create a new project when an invalid project.conf is spotted, and indeed we should probably be seeing an invalid YAML error here, the "mapping values are not allowed here" is not very handy, will have to investigate whether this is coming from BuildStream or from `ruamel.yaml`.

Thanks a lot for elaborating [[Gitlab user @pointswaves]](https://gitlab.com/pointswaves)

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @coldtom]](https://gitlab.com/coldtom) on May 9, 2019, 13:33

This looks like a duplicate of #642, so a backport of !792 should be a solution.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @coldtom]](https://gitlab.com/coldtom) on May 9, 2019, 14:47

mentioned in merge request !1338

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on May 13, 2019, 09:05

Thanks for backporting this [[Gitlab user @coldtom]](https://gitlab.com/coldtom), this has been merged in `bst-1.2` and should be fixed in the upcoming `1.2.7` release.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on May 13, 2019, 09:05

closed
